cips/cip-add-hyperlane.md

```diff
@@ -0,0 +1,59 @@
+| cip           | xx                                                                                         |
+|---------------|--------------------------------------------------------------------------------------------|
+| title         | Add Hyperlane to celestia-app                                                              |
+| description   | This CIP proposes adding Hyperlane to celestia-app.                                        |
+| author        | c-node [@c-node](https://github.com/S1nus)                                                 |
+| discussions-to | [forum](https://forum.celestia.org/t/cip-add-hyperlane-bridging/1909)                     |                                                                     |
+| status        | Idea                                                                                      |
+| type          | Standards Track                                                                            |
+| category      | Core                                                                                       |
+| created       | 2025-02-21                                                                                `|
+
+## Abstract
+
+We propose adding Hyperlane functionality to celestia-app, either through whitelisted cosmWASM, through the Hyperlane cosmos-sdk module, or some other way.
+
+## Motivation
+
+Hyperlane is a popular bridging protocol for the TIA token. Currently, users have to take inconvenient and indirect routes through other chains to use Hyperlane, since Celestia does not include it natively.
+
+
+## Specification
+
+TBD
+
+### Overview
+
+TBD
+
+### Detailed Design
+
+TBD
+
+## Parameters
+
+TBD
+
+## Rationale
+
+TBD
+
+## Backwards Compatibility
+
+TBD
+
+## Test Cases
+
+TBD
```

Comment:
Proposal: add a test case that entails deploying Hyperlane on a Celestia testnet and establish a Hyperlane bridge from another Cosmos / EVM testnet to Celestia testnet. Then send TIA from Celestia to remote chain and verify it works. Then send a non TIA token from remote chain to Celestia and verify it doesn't work. Note: this assumes that we add something like the tokenfilter to the Hyperlane Cosmos SDK module on Celestia.